core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -275,6 +285,23 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
         return Optional.of(tableScan);
     }
 
+    // PushProjectionIntoTableScan might be executed after AddExchanges and DetermineTableScanNodePartitioning.
+    // Table scan node partitioning should not change after these optimizers are executed since that would make with
+    // exchanges added invalid.
+    private static void verifyTablePartitioning(
+            Session session,
+            Metadata metadata,
+            TableScanNode oldTableScan,
+            Optional<TablePartitioning> newTablePartitioning)
+    {
+        if (oldTableScan.getUseConnectorNodePartitioning().isEmpty()) {
```

Comment:
I guess I must be confused then about what your comment about
```
    // Table scan node partitioning should not change after these optimizers are executed since that would make with
    // exchanges added invalid.
```
actually means for the developer then. I thought it meant to indicate that this rule should always encounter a non-empty node partitioning intent. Maybe we could clarify this a bit more in the comment?